Automattic/newspack-plugin
# feat: Add popup events to GA4

### All Submissions:






### Changes proposed in this Pull Request:



Connects the Popup events to GA4.

Note that the Popup events have already been tested. This only connects them to GA4. It also adds a little more detail to the newsletter subscription events.

This PR also fixes a bug in the `donation_new` event and the `form_submission_success` event for donations. Both when using Stripe as platform.

Since the new donation in Stripe is triggered when Stripe server hits our webhook, the GA Client and Session IDs were not bein fetched by the cookie. So I had to add these information in the payment metadata and retrieve them for the order when they are marked as completed.

### How to test the changes in this Pull Request:

Since we changed the structure of the data and also fixed a bug, we should do an extensive tests of as many events as possible.

Get the GA4 credentials. You will need your measurement ID and an API secret.

* api_secret - Required. An API SECRET generated in the Google Analytics UI. To create a new secret, navigate to:
    Admin > Data Streams > choose your stream > Measurement Protocol > Create
* measurement_id - Required. The measurement ID associated with a stream. Found in the Google Analytics UI under:
    Admin > Data Streams > choose your stream > Measurement ID

Store this info in the database:
```
wp option set ga4_measurement_id "G-XXXXXXXXXX"
wp option set ga4_api_secret YYYYYYYYYYYYYYYYYYY
```
Now, let's test:

1. Make sure RAS is active and that you have a registered reader
2. Make sure to add this constant to wp-config
```
define( 'NEWSPACK_LOG_LEVEL', 2 );
define( 'NEWSPACK_EXPERIMENTAL_GA4_EVENTS', true );
```
4. Checkout the newsletters plugin on https://github.com/Automattic/newspack-newsletters/pull/1111
5. Setup a popup with a Newsletter subscription form
6. Subscribe to a newsletter via this popup
8. Visit the Google Analytics Realtime dashboard
9. See the `campaign_interaction` events coming in

![Captura de tela de 2023-03-09 13-15-59](https://user-images.githubusercontent.com/971483/224086546-fc9ee3ac-a5e7-4a08-9921-7c00d78b5ed5.png)

11. Inspect them and see check its attributes to see if the values are correct.

![Captura de tela de 2023-03-09 13-51-25](https://user-images.githubusercontent.com/971483/224097513-bf0700aa-c552-4e66-bc8e-358fd1c08279.png)

### Other information:
